fireplace/dsl/random_picker.py

```diff
@@ -1,79 +1,97 @@
+		else:
+			newFilters = filters.copy()
 
-	def get_cards(self, source):
-		filters = self.filters.copy()
-		# Iterate through the filters, evaluating the LazyValues as we go
-		for k, v in filters.items():
+		for k, v in newFilters.items():
 			if isinstance(v, LazyValue):
-				filters[k] = v.evaluate(source)
-		return self._filter_cards(filters)
+				newFilters[k] = v.evaluate(source)
+
+		return cards.filter(**newFilters)
+
+	# This picks from a single combined card pool without replacement, weighting each filtered set of cards against the total
+	def evaluate(self, source, cards=None) -> str:
+
+		# use specific card list if given
+		if cards:
+			self.weights = [1]
+			card_sets = [cards]
 
-	def evaluate(self, source) -> str:
-		if self.lazy_filters:
-			# If the card has lazy filters, we need to evaluate them
-			cards = self.get_cards(source)
+		# use global filters if no weighted filter sets given
+		elif not self.weightedfilters:
+			self.weights = [1]
+			card_sets = [self.find_cards(source)]
+
+		# otherwise find cards for each set of filters
 		else:
-			cards = self.cards
-		ret = random.sample(cards, self.count)
-		return [source.controller.card(card, source=source) for card in ret]
-
-
-RandomCard = lambda *a, **kw: RandomCardPicker(*a, **kw)
-RandomCollectible = lambda *a, **kw: RandomCardPicker(*a, collectible=True, **kw)
-RandomMinion = lambda *a, **kw: RandomCollectible(*a, type=CardType.MINION, **kw)
-RandomBeast = lambda *a, **kw: RandomMinion(*a, race=Race.BEAST)
-RandomMech = lambda *a, **kw: RandomMinion(*a, race=Race.MECHANICAL)
-RandomMurloc = lambda *a, **kw: RandomMinion(*a, race=Race.MURLOC)
-RandomSpell = lambda *a, **kw: RandomCollectible(*a, type=CardType.SPELL, **kw)
-RandomTotem = lambda *a, **kw: RandomCardPicker(*a, race=Race.TOTEM)
-RandomWeapon = lambda *a, **kw: RandomCollectible(*a, type=CardType.WEAPON, **kw)
-RandomLegendaryMinion = lambda *a, **kw: RandomMinion(*a, rarity=Rarity.LEGENDARY, **kw)
+			# add the global filters to each set of filters
+			wf = [{ **x, **self.globalfilter } for x in self.weightedfilters]
+			card_sets = [self.find_cards(source, **x) for x in wf]
+
+		# get weighted sample of card pools
+		from ..cards.utils import weighted_card_choice
+		return weighted_card_choice(source, self.weights, card_sets, self.count)
+
+
+RandomCard = lambda **kw: RandomCardPicker(**kw)
+RandomCollectible = lambda **kw: RandomCardPicker(collectible=True, **kw)
+RandomMinion = lambda **kw: RandomCollectible(type=CardType.MINION, **kw)
+RandomBeast = lambda **kw: RandomMinion(race=Race.BEAST)
+RandomMech = lambda **kw: RandomMinion(race=Race.MECHANICAL)
+RandomMurloc = lambda **kw: RandomMinion(race=Race.MURLOC)
+RandomSpell = lambda **kw: RandomCollectible(type=CardType.SPELL, **kw)
+RandomTotem = lambda **kw: RandomCardPicker(race=Race.TOTEM)
+RandomWeapon = lambda **kw: RandomCollectible(type=CardType.WEAPON, **kw)
+RandomLegendaryMinion = lambda **kw: RandomMinion(rarity=Rarity.LEGENDARY, **kw)
```

Comment:
Are you 100% certain random pickers will no longer take positional arguments? does this even need to be done?